src/emqx_protocol.erl

```diff
@@ -421,7 +427,95 @@ process_packet(?DISCONNECT_PACKET(?RC_SUCCESS, #{'Session-Expiry-Interval' := In
 process_packet(?DISCONNECT_PACKET(?RC_SUCCESS), PState) ->
     {stop, normal, PState#pstate{will_msg = undefined}};
 process_packet(?DISCONNECT_PACKET(_), PState) ->
-    {stop, normal, PState}.
+    {stop, normal, PState};
+
+process_packet(?AUTH_PACKET(_), PState = #pstate{auth_state = disconnected}) ->
+    {error, proto_not_connected, PState};
+process_packet(?AUTH_PACKET(_), PState = #pstate{auth_method = ?NO_AUTH_METHOD, auth_state = AuthState}) ->
+    if 
+        AuthState =:= authenticating ->
+            connack({?RC_PROTOCOL_ERROR, PState});
+        true ->     % reauthenticating or connected 
+            disconnect({?RC_PROTOCOL_ERROR, PState})
+    end;
+process_packet(?AUTH_PACKET(_, undefined), PState = #pstate{auth_state = AuthState}) ->
+    if 
+        AuthState =:= authenticating ->
+            connack({?RC_MALFORMED_PACKET, PState});
+        true ->     % reauthenticating or connected 
+            disconnect({?RC_MALFORMED_PACKET, PState})
+    end;
+process_packet(?AUTH_PACKET(?RC_CONTINUE_AUTHENTICATE, _), PState = #pstate{auth_state = connected}) -> 
+    disconnect({?RC_PROTOCOL_ERROR, PState});
+process_packet(?AUTH_PACKET(?RC_CONTINUE_AUTHENTICATE, Properties), PState) ->
+    process_enhanced_auth(maps:get('Authentication-Method', Properties, ?NO_AUTH_METHOD), maps:get('Authentication-Data', Properties, ?NO_AUTH_DATA), PState);
```

Comment:
Change `maps:get('Authentication-Method', Properties, ?NO_AUTH_METHOD)` to `auth_method(Properties)`